### PR TITLE
Querying tags

### PR DESCRIPTION
**Pull request by willsb**
Ref #518 

There’s _some_ UI feedback, though one still can’t select tags.

Tests are missing, but feel free to give an overall look at it to see if it looks good as is.

### CODE REVIEW

File: Toggl.Foundation/Autocomplete/TextFieldInfo.cs [hunk 1]
```diff
@@ -1,9 +1,14 @@
 ﻿using System;
+using System.Collections.Generic;
+using Toggl.Foundation.Autocomplete.Suggestions;
+using Toggl.Multivac.Models;
```
Comment (arturs105) [inline Toggl.Foundation/Autocomplete/TextFieldInfo.cs]: 🔥

File: Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs [hunk 2]
```diff
@@ -255,16 +267,71 @@ public sealed class QueriesTheDatabaseForProjects : AutocompleteProviderTest
...
+                [Fact]
+                public async Task SuggestsAllTagstWhenThereIsNoStringToSearch()
+                {
+                    const string description = "#";
+                    var textFieldInfo = TextFieldInfo.Empty.WithTextAndCursor(description, 1);
+
+                    var suggestions = await Provider.Query(textFieldInfo);
+
+                    await Database.Tags.Received().GetAll();
+                    suggestions.Should().HaveCount(10)
```
Comment (arturs105) [inline Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs]: `.HaveCount(Tags.Count())`

File: Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs [hunk 3]
```diff
@@ -255,16 +267,71 @@ public sealed class QueriesTheDatabaseForProjects : AutocompleteProviderTest
...
+                public async Task SearchesTheName()
+                {
+                    const string description = "#40";
+                    var textFieldInfo = TextFieldInfo.Empty.WithTextAndCursor(description, 1);
+
+                    var suggestions = await Provider.Query(textFieldInfo);
+
+                    await Database.Tags.Received().GetAll();
+                    suggestions.Should().HaveCount(1)
+                        .And.AllBeOfType<TagSuggestion>();
```
Comment (arturs105) [inline Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs]: Could also test, whether the returned tag actually has name "40".

File: Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs [hunk 4]
```diff
@@ -255,16 +267,71 @@ public sealed class QueriesTheDatabaseForProjects : AutocompleteProviderTest
...
+                    var actualDescription = $"{description}#{description}";
+                    var textFieldInfo = TextFieldInfo.Empty.WithTextAndCursor(actualDescription, description.Length + 1);
+
+                    await Provider.Query(textFieldInfo);
+
+                    await Database.Tags.Received().GetAll();
+                }
+
+                [Fact]
+                public async Task SuggestsAllTagstWhenThereIsNoStringToSearch()
```
Comment (arturs105) [inline Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs]: Typo: SuggestsAllTags**t**WhenThereIsNoStringToSearch

File: Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs [hunk 5]
```diff
@@ -255,16 +267,71 @@ public sealed class QueriesTheDatabaseForProjects : AutocompleteProviderTest
...
+
+                    var suggestions = await Provider.Query(textFieldInfo);
+
+                    await Database.Tags.Received().GetAll();
+                    suggestions.Should().HaveCount(1)
+                        .And.AllBeOfType<TagSuggestion>();
+                }
+
+                [Fact]
+                public async Task OnlyDisplaysResultsTheHaveHasAtLeastOneMatchOnEveryWordTyped()
```
Comment (arturs105) [inline Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs]: Was this meant "Only displays results that have at least one match on every word typed"?

File: Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs (hunk 4 above)
Comment (willsb) [reply Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs]: Whomst'd've

**Pull request merged**